dbt/include/fabric/macros/adapters/metadata.sql

```diff
@@ -9,6 +9,18 @@
   information_schema
 {%- endmacro %}
 
+{% macro get_use_database_sql(database) %}
+    {{ return(adapter.dispatch('get_use_database_sql', 'dbt')(database)) }}
+{% endmacro %}
+
+{%- macro default__get_use_database_sql(database) -%}
+{%- endmacro -%}
+
+
+{%- macro fabric__get_use_database_sql(database) -%}
+  USE [{{database}}];
```

Comment:
It looks like you're always passing a `Relation` object into this macro. That signals to me that this macro should take a `Relation` argument instead of a string:
```
{%- macro fabric__get_use_database_sql(relation) -%}
  USE [{{ relation.database }}];
{%- endmacro -%}
```